x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/es_index.tsx

```diff
@@ -38,33 +70,165 @@ export function getActionType(): ActionTypeModel {
+            defaultMessage="Use * to broaden your query."
+          />
+        }
+      >
+        <EuiComboBox
+          fullWidth
+          singleSelection={{ asPlainText: true }}
+          async
+          isLoading={isIndiciesLoading}
+          isInvalid={errors.index.length > 0 && index !== undefined}
+          noSuggestions={!indexOptions.length}
+          options={indexOptions}
+          data-test-subj="connectorIndexesComboBox"
+          selectedOptions={
+            index
+              ? [
+                  {
+                    value: index,
+                    label: index,
+                  },
+                ]
+              : []
           }
+          onChange={async (selected: EuiComboBoxOptionOption[]) => {
+            editActionConfig('index', selected[0].value);
+            const indices = selected.map(s => s.value as string);
+
+            // reset time field and expression fields if indices are deleted
+            if (indices.length === 0) {
+              setTimeFieldOptions([]);
+              return;
+            }
+            const currentEsFields = await getFields(http!, indices);
+            const timeFields = getTimeFieldOptions(currentEsFields as any);
+
+            setTimeFieldOptions([firstFieldOption, ...timeFields]);
+          }}
+          onSearchChange={async search => {
+            setIsIndiciesLoading(true);
+            setIndexOptions(await getIndexOptions(http!, search, indexPatterns));
+            setIsIndiciesLoading(false);
+          }}
+          onBlur={() => {
+            if (!index) {
+              editActionConfig('index', '');
+            }
+          }}
+        />
+      </EuiFormRow>
+      <EuiSpacer size="m" />
+      <EuiSwitch
+        data-test-subj="indexRefreshCheckbox"
+        checked={refresh || false}
+        onChange={e => {
+          editActionConfig('refresh', e.target.checked);
         }}
+        label={
+          <FormattedMessage
+            id="xpack.triggersActionsUI.components.builtinActionTypes.indexAction.refreshLabel"
+            defaultMessage="Refresh"
+          />
+        }
+      />
+      <EuiSpacer size="m" />
+      <EuiSwitch
+        data-test-subj="hasTimeFieldCheckbox"
+        checked={hasTimeFieldCheckbox || false}
+        onChange={() => {
+          setTimeFieldCheckboxState(!hasTimeFieldCheckbox);
+        }}
+        label={
+          <FormattedMessage
+            id="xpack.triggersActionsUI.components.builtinActionTypes.indexAction.defineTimeFieldLabel"
+            defaultMessage="Define time field"
+          />
+        }
       />
-    </EuiFormRow>
+      <EuiSpacer size="m" />
+      {hasTimeFieldCheckbox ? (
+        <>
+          <EuiFormRow
+            id="thresholdTimeField"
+            fullWidth
+            label={
+              <FormattedMessage
+                id="xpack.triggersActionsUI.components.builtinActionTypes.indexAction.timeFieldLabel"
+                defaultMessage="Timefield"
```

Comment:
```suggestion
                defaultMessage="Time field"
```

Reply:
Did as on mockup, but yes, agree that better to split.
![Img](https://i.imgur.com/28GK1Jx.png)